tests/test_fieldset_sampling.py

```diff
@@ -426,27 +426,29 @@ def test_list_of_fields(mode, with_W, k_sample_p, mesh):
                mesh=mesh, fieldtype='U')
     V1 = Field('V1', np.zeros((zdim*gf, ydim*gf, xdim*gf), dtype=np.float32), grid=U1.grid, fieldtype='V')
     V2 = Field('V2', np.zeros((zdim, ydim, xdim), dtype=np.float32), grid=U2.grid, fieldtype='V')
-    fieldset = FieldSet([U1, U2], [V1, V2])
+    fieldset = FieldSet(U1+U2, V1+V2)
 
     conv = 1852*60 if mesh == 'spherical' else 1.
     assert np.allclose(fieldset.U.eval(0, 0, 0, 0)*conv, 0.3)
     assert np.allclose(fieldset.U[0, 0, 0, 0]*conv, 0.3)
 
     P1 = Field('P1', 30*np.ones((zdim*gf, ydim*gf, xdim*gf), dtype=np.float32), grid=U1.grid)
     P2 = Field('P2', 20*np.ones((zdim, ydim, xdim), dtype=np.float32), grid=U2.grid)
-    fieldset.add_field([P1, P2], name='P')
-    assert np.allclose(fieldset.P[0, 0, 0, 0], 50)
+    P3 = Field('P3', 10*np.ones((zdim, ydim, xdim), dtype=np.float32), grid=U2.grid)
+    P4 = Field('P4', 0*np.ones((zdim, ydim, xdim), dtype=np.float32), grid=U2.grid)
+    fieldset.add_field((P1+P4)+(P2+P3), name='P')
```

Comment:
So adding fields is neither commutative (was already the case) nor associative, right?

Reply:
Cool, I see it is now associative since the change on Field.__add__
By 'not commutative', I was only referring that FA + FB creates [FA, FB] and FB + FA creates [FB, FA]
But of course the result of the interpolation is the same at the end.